mopidy_musicbox_webclient/static/js/images.js

```diff
@@ -4,105 +4,279 @@
     } else if (typeof module === 'object' && module.exports) {
         module.exports = factory()
     } else {
-        root.coverArt = factory()
+        root.images = factory()
     }
 }(this, function () {
     'use strict'
 
     var API_KEY = 'b6d34c3af91d62ab0ae00ab1b6fa8733'
     var API_SECRET = '2c631802c2285d5d5d1502462fe42a2b'
 
-    var coverArt = {
-        lastfm: new LastFM({
+    var images = {
+
+        DEFAULT_ALBUM_URL: 'images/default_cover.png',
+        DEFAULT_ARTIST_URL: 'images/user_24x32.png',
+
+        lastFM: new LastFM({
             apiKey: API_KEY,
             apiSecret: API_SECRET,
             cache: new LastFMCache()
         }),
 
-        getCover: function (uri, images, size, mopidy) {
-            var defUrl = 'images/default_cover.png'
-            $(images).attr('src', defUrl)
+        /* Extract artist information from Mopidy track. */
+        _getArtistInfo: function (track) {
+            var artistName = ''
+            var musicBrainzID = ''
+
+            if (track && track.artists && (track.artists.length > 0)) {
+                // First look for the artist info in the track
+                artistName = track.artists[0].name
+                musicBrainzID = track.artists[0].musicbrainz_id
+            }
+
+            if ((!artistName || !musicBrainzID) && (track && track.album && track.album.artists && track.album.artists.length > 0)) {
+                // Fallback to using artist info contained in the track's album
+                artistName = artistName || track.album.artists[0].name
+                musicBrainzID = musicBrainzID || track.album.artists[0].musicbrainz_id
+            }
+
+            return {mbid: musicBrainzID, name: artistName}
+        },
+
+        /* Utility function for retrieving artist informaton for the given track from last.fm */
+        _getLastFmArtistInfo: function (track) {
+            var artist = images._getArtistInfo(track)
+            var artistPromise = $.Deferred()
+
+            if (!track || (!track.musicbrainz_id && !track.name) || (!track.musicbrainz_id && (!artist || !artist.name))) {
```

Comment:
Yes. Isn't that what this if testing for? It is confusing because it is stated in the negative, but I read it as:

1. we need a track, and then
2. we need either the mbid, or the track name
3. if we do not have mbid, then we need the artist name as well

So we should then be able to do a lookup using either ``mbid``, or ``track.name`` **AND** ``artist.name``.